resources/snakefiles/Snakefile.smk

```diff
@@ -0,0 +1,242 @@
+        bedtools bamtofastq -i {output.temp_dir}/{wildcards.sample}.mapped.bam \
+          -fq {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq \
+          -fq2 {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq \
+          2>> {log.mapped}
+
+        # zip the filtered fastqs
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R1.trimmed.filtered.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R1.trimmed.filtered.fastq.gz
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R2.trimmed.filtered.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R2.trimmed.filtered.fastq.gz
+
+        # zip the host fastqs
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq.gz
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq.gz
+
+        # copy the filtered fastqs to final location
+        cp {output.temp_dir}/{wildcards.sample}.R1.trimmed.filtered.fastq.gz \
+          {output.nonhost_R1}
+        cp {output.temp_dir}/{wildcards.sample}.R2.trimmed.filtered.fastq.gz \
+          {output.nonhost_R2}
+
+        # copy the host fastqs to final location
+        cp {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq.gz \
+          {output.host_R1}
+        cp {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq.gz \
+          {output.host_R2}
+        """
+
+
+rule metaspades_assembly:
+    """
+
+    Performs a metagenomic assembly on a sample using MetaSPAdes.
+
+    """
+    input:
+        fastq1=rules.host_filter.output.nonhost_R1,
+        fastq2=rules.host_filter.output.nonhost_R2
+    output:
+        contigs="output/metaspades/{sample}.contigs.fasta",
+        temp_dir=temp(directory("output/{sample}_temp"))
+    conda:
+        "resources/envs/spades.yaml"
+    threads:
+        config['threads']['assembly']
+    benchmark:
+        "output/benchmarks/metaspades/{sample}_benchmark.txt"
+    shell:
+        """
+        # Make temporary output directory
+        mkdir -p {output.temp_dir}
+
+        # Make directory for assemblies and log files
+        mkdir -p output/metaspades/
+        mkdir -p output/logs/metaspades/
+
+        # run the metaspades assmebly
+        metaspades.py --threads {threads} \
+          -o output/{output.temp_dir}/ \
+          --pe1-1 {input.fastq1} \
+          --pe1-2 {input.fastq2}
+
+        # move and rename the contigs file into a permanent directory
+        mv output/{output.temp_dir}/contigs.fasta output/metaspades/{wildcards.sample}.contigs.fasta
+
+        # move the log file
+        mv output/{output.temp_dir}/spades.log output/logs/metaspades/{wildcards.sample}_log.txt
+
+        # remove the temp directory
```

Comment:
We don't want to do this manually. By specifying the output with the `temp()` function, Snakemake will handle removal automatically. This also allows us to easily later add downstream rules that use one of the files from the temp output directory as an input -- if Snakemake recognizes that there are downstream rules requiring an output specified with `temp()`, it will keep that output around until those downstream rules have all been fulfilled before deleting!

Reply:
This is how I thought it worked, but it wasn't being deleted! I'm not sure why, but maybe there is another reason that Snakemake was keeping it around. When I tried to explicitly delete it with `rm -rf`, Snakemake complained that it was missing an output. I'll work on it more though.